File: revese integer/ReverseInteger.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if(x>=0):
            ret = int(str(x)[::-1],10)
        else:
            ret = 0 - int(str(x)[1::][::-1],10)
        print(ret)
        print((1<<31-1))
        if(((0-(1<<31))>= ret) or (((1<<31)-1)<= ret)):
            return 0
        else:
            return ret

File: revese integer/ReverseInteger.c
#  define INT_MIN   (-INT_MAX - 1)
#  define INT_MAX   2147483647

int reverse(int x){
    long ret = 0;
    
      
    while(x!=0)
    {
        ret = ret*10 + (x%10);
        x = x/10;
        if( (ret < INT_MIN) || (ret > INT_MAX))
        {
            ret = 0;
            break;
        }
    }
    ret = (int)ret;
    return ret;
 
}


